nmoscommon/mdns/mdnsEngine.py

```diff
@@ -101,10 +102,12 @@ def unregister(self, name, regType):
 
     def callback_on_services(self, regtype, callback, registerOnly=True, domain=None):
         self._autostart_if_required()
-        if domain == "local" or domain is None:
+        doDNSDiscover = domain != 'local' and config['dns_discover']
```

Comment:
I wonder if we can ignore the 'domain' variable here entirely. This avoids an accidental misconfiguration in the config file of 'domain': 'local' and 'mdns_discover': false, which would result in no discovery happening at all. That way 'domain' is only ever used to refer to a unicast domain.

At present the behaviour looks good, the only issue is that I believe it's still not possible to manually configure a domain, but have mDNS discovery continue to operate. Ignoring the 'domain' in this logic should permit that mode of operation.